cmd/login/runner.go

```diff
@@ -0,0 +1,234 @@
+package login
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	gooidc "github.com/coreos/go-oidc"
+	"github.com/fatih/color"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/skratchdot/open-golang/open"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+
+	"github.com/giantswarm/kubectl-gs/cmd/login/template"
+	"github.com/giantswarm/kubectl-gs/pkg/callbackserver"
+	"github.com/giantswarm/kubectl-gs/pkg/installation"
+	"github.com/giantswarm/kubectl-gs/pkg/kubeconfig"
+	"github.com/giantswarm/kubectl-gs/pkg/oidc"
+)
+
+const (
+	clientID     = "zQiFLUnrTFQwrybYzeY53hWWfhOKWRAU"
+	clientSecret = "TVHzVPin2WTiCma6bqp5hdxgKbWZKRbz" // #nosec G101
```

Comment:
What are these values?

Reply:
I meant how they are generated? are static? can we expose them in code?
I imagine they are sued for oidc client, but I dont know exactly how it works

Reply:
Yes, these are for OIDC. They are static and I've generated them using some random online tool.
We're using a dex [public client](https://github.com/dexidp/dex/blob/master/Documentation/custom-scopes-claims-clients.md#public-clients), so exposing these credentials is not a security issue